client/src/dataset/components/ScanSizePart.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from "react";
+
+interface ScanSizePartProps {
+    name: string,
+    id: string,
+    scanKey: number,
+    value: number,
+    scanSizeChangeHandle: (idx: number, value: string) => void,
+    commaPressHandle: (idx: number, keyCode: number) => void,
+    scanRef: (ref:HTMLInputElement) => void,
+}
+
+const ScanSizePart: React.FC<ScanSizePartProps> = ({ name, id, scanKey, scanSizeChangeHandle, value, commaPressHandle, scanRef }) => {
+   const onPartChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      scanSizeChangeHandle(scanKey, e.target.value);
+   }
+   const onCommaPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+      commaPressHandle(scanKey, e.keyCode);
+   }
```

Comment:
You might want to move the keycode check here, otherwise the name of the callback, `commaPressHandle`, is a bit misleading, as it behaves like `onKeyDown`